clients/ts/signalr/tests/WebSocketTransport.test.ts

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { ILogger } from "../src/ILogger";
+import { TransferFormat } from "../src/ITransport";
+import { NullLogger } from "../src/Loggers";
+import { WebSocketTransport } from "../src/WebSocketTransport";
+import { VerifyLogger } from "./Common";
+import { TestMessageEvent } from "./TestEventSource";
+import { TestCloseEvent, TestEvent, TestWebSocket } from "./TestWebSocket";
+
+describe("WebSocketTransport", () => {
+    [[TransferFormat.Text, "blob"],
+    [TransferFormat.Binary, "arraybuffer"]]
+        .forEach(([input, expected]) => {
+            it(`sets websocket binarytype based on transferformat: ${TransferFormat[input]}`, async () => {
+                await VerifyLogger.run(async (logger) => {
+                    const webSocket = new WebSocketTransport(undefined, logger, true, TestWebSocket);
+
+                    const connectPromise = webSocket.connect("http://example.com", input as TransferFormat);
+
+                    await TestWebSocket.webSocket.openSet;
+                    TestWebSocket.webSocket.onopen(new TestEvent());
+
+                    await connectPromise;
+
+                    expect(TestWebSocket.webSocket.binaryType).toEqual(expected);
+                });
+            });
+        });
+
+    it("connect waits for WebSocket to be connected", async () => {
+        const webSocket = new WebSocketTransport(undefined, NullLogger.instance, true, TestWebSocket);
+
+        let connectComplete: boolean = false;
+        const connectPromise = (async () => {
+            await webSocket.connect("http://example.com", TransferFormat.Text);
+            connectComplete = true;
+        })();
+
+        await TestWebSocket.webSocket.openSet;
+
+        expect(connectComplete).toEqual(false);
+
+        TestWebSocket.webSocket.onopen(new TestEvent());
+
+        await connectPromise;
+        expect(connectComplete).toEqual(true);
+    });
+
+    it("connect fails if there is error during connect", async () => {
+        (global as any).ErrorEvent = TestEvent;
+        const webSocket = new WebSocketTransport(undefined, NullLogger.instance, true, TestWebSocket);
+
+        let connectComplete: boolean = false;
+        const connectPromise = (async () => {
+            await webSocket.connect("http://example.com", TransferFormat.Text);
+            connectComplete = true;
+        })();
+
+        await TestWebSocket.webSocket.openSet;
+
+        expect(connectComplete).toEqual(false);
+
+        TestWebSocket.webSocket.onerror(new TestEvent());
+
+        await expect(connectPromise)
+            .rejects;
+        expect(connectComplete).toEqual(false);
+    });
+
+    [["http://example.com", "ws://example.com?access_token=secretToken"],
+    ["http://example.com?value=null", "ws://example.com?value=null&access_token=secretToken"]]
```

Comment:
Throw some tests in that don't use an access_token to also verify the conversion from `http(s)` to `ws(s)`

Reply:
Maybe two separate "theories" (one with an access_token and one without) so the names don't get too funky:

* `generates correct WebSocket URL for ${input}`
* `generates correct WebSocket URL for ${input} with access_token`